app/models/concerns/hostext/operating_system.rb

```diff
@@ -34,7 +34,8 @@ def available_template_kinds(provisioning = nil)
 
     def template_kinds(provisioning = nil)
       return TemplateKind.all unless provisioning == 'image'
-      cr     = ComputeResource.find_by_id(compute_resource_id)
+      cr_id  = compute_resource_id || hostgroup.compute_resource_id
```

Comment:
A host doesn't need to have a hostgroup, right?
```suggestion
      cr_id  = compute_resource_id || hostgroup&.compute_resource_id
```

Reply:
@ekohl  correct :)